tests/integration-service/integration.go

```diff
@@ -201,6 +217,18 @@ var _ = framework.IntegrationServiceSuiteDescribe("Integration Service E2E tests
 			Expect(f.AsKubeDeveloper.IntegrationController.WaitForAllIntegrationPipelinesToBeFinished(testNamespace, applicationName, snapshot)).To(Succeed())
 		})
 
+		It("checks if the status of integration test si reported in the Snapshot", func() {
+			snapshot, err = f.AsKubeDeveloper.IntegrationController.WaitForSnapshotToGetCreated("", pipelineRun.Name, componentName, testNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			statuses, err := f.AsKubeDeveloper.IntegrationController.GetIntegrationTestsStatusesFromSnapshot(snapshot)
+			Expect(err).ToNot(HaveOccurred())
+
+			statusDetail, ok := statuses.GetScenarioStatus(integrationTestScenario.Name)
+			Expect(ok).To(BeTrue())
+			Expect(statusDetail.Status).To(Equal(intgteststat.IntegrationTestStatusTestFail))
```

Comment:
The only concern I have is: what if the status doesn't get updated faster before this line is executed by the test?
To test a situation like this, we can run this test 5-6 times to see if it fails or not. If it does, we should add this logic within an Eventually block.

But the question is: are you able to run this test locally?

Reply:
I can add it into eventually block.

I haven't tested it locally, I was not able to run tests due to random issues with dev clusters

Reply:
Then let's add it to an Eventually block just to be safe.

Reply:
Added, I reworked it little bit (from code structure POW, test is the same)

Reply:
Thanks!